src/region/region.c

```diff
@@ -35,117 +48,127 @@ enum st_region_state_e {
 
 
 static inline int
-is_reg_in_state(const st_region_reg_t *reg_ptr, uint64_t state)
+is_reg_in_state(const st_region_reg_t *reg, uint64_t state)
 {
-    return (reg_ptr->state == state);
+    return (reg->state == state);
 }
 
 static inline int
-is_reg_free(const st_region_reg_t *reg_ptr)
+is_reg_state_free(const st_region_reg_t *reg)
 {
-    return is_reg_in_state(reg_ptr, ST_REGION_STATE_FREE);
+    return is_reg_in_state(reg, ST_REGION_STATE_FREE);
 }
 
 static inline int
-is_reg_busy(const st_region_reg_t *reg_ptr)
+is_reg_state_busy(const st_region_reg_t *reg)
 {
-    return is_reg_in_state(reg_ptr, ST_REGION_STATE_BUSY);
+    return is_reg_in_state(reg, ST_REGION_STATE_BUSY);
 }
 
-void *
-st_region_create(void *addr, uint32_t length,
-                 uint32_t prot, uint32_t flags, int *ret_shm_fd)
+int
+st_region_shm_create(uint32_t length, void **ret_addr, int *ret_shm_fd)
 {
+    st_must(ret_shm_fd != NULL, ST_ARG_INVALID);
+    st_must(ret_addr != NULL, ST_ARG_INVALID);
+
+    int ret    = -1;
     int shm_fd = shm_open(ST_REGION_SHM_OBJ_PATH,
-                          O_CREAT|O_RDWR|O_TRUNC,
+                          O_CREAT | O_RDWR | O_TRUNC,
                           ST_REGION_SHM_OBJ_MODE);
     if (shm_fd == -1) {
         derrno("failed to shm_open");
 
-        return MAP_FAILED;
+        return errno;
```

Comment:
前面的derrno是不是会重写errno呢，
用不用在derrno之前将errno保存一下，然后再return呢

Reply:
有一点点小风险：
The  XSI-compliant  strerror_r()  function  returns  0  on  success.  On error, a (positive) error number is
returned (since glibc 2.13), or -1 is returned and errno is set to indicate the error (glibc versions before
2.13).

在strerror内部出错的时候是可能修改errno的。

an "Unknown error nnn" message if the error number is unknown.
文档上没说什么情况会返回错误，我调整一下代码

Reply:
有一点点小风险：
The XSI-compliant strerror_r() function returns 0 on success. On error, a (positive) error number is
returned (since glibc 2.13), or -1 is returned and errno is set to indicate the error (glibc versions before
2.13).

在strerror内部出错的时候是可能修改errno的。

an "Unknown error nnn" message if the error number is unknown.
文档上没说什么情况会返回错误。

还有就是咱们的代码到了product级别编译的时候，这些输出log的函数会被替换成空，所以这里应该没影响的

Reply:
哦哦